tests/foreman/api/test_bookmarks.py

```diff
@@ -0,0 +1,290 @@
+
+        @Assert: The new bookmark name is listed
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_name(self):
+        """Update a bookmark with name already taken
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create 2 bookmarks of random names with random query
+
+        @Steps:
+
+        1. Update the name to one of the other pre-created bookmarks
+
+        @Assert: Error - name already taken, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_name_empty(self):
+        """Update a bookmark with an empty name
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the name of the pre-created bookmark to an empty value
+
+        @Assert: Error - name cannot be empty, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_positive_update_bookmark_query(self):
+        """Update a bookmark query
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the query of the pre-created bookmark
+
+        @Assert: The updated query submitted
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_query_empty(self):
+        """Update a bookmark with an empty query
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the query of the pre-created bookmark
+
+        @Assert: Error - search query cannot be empty, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_positive_update_bookmark_public(self):
+        """Update a bookmark public state
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create 2 bookmarks of a random names with random queries, one public
+        and one private
+        2. Create a non-admin user with 'viewer' role
+
+        @Steps:
+
+        1. List the bookmarks using the pre-created user's credentials
```

Comment:
@omaciel after inspecting the `private/public` feature in more detail, It makes no sense to create an additional user. The `private/public` feature does not have an effect on the common bookmark listing, it only applies to visibility at the entity page which is not applicable for the `API`.
this test should be replaced by 2 additional tests:
`test_positive_update_public`
`test_positive_update_private`
where only updating the state would be tested and asserted.